on:
  issue_comment:
    types:
      - created
jobs:
  container-tests:
    runs-on: ubuntu-22.04
    name: "Timeout test for testing farm as a github action"

    if: |
      github.event.issue.pull_request
      && contains(github.event.comment.body, '[test]')
      && contains(fromJson('["OWNER", "MEMBER"]'), github.event.comment.author_association)

    steps:
      - name: Checkout repo
        uses: actions/checkout@v5
        with:
          ref: "refs/pull/${{ github.event.issue.number }}/head"

      - name: Run the tests
        uses: ./
        id: tf_results
        with:
          api_key: ${{ secrets.TF_PUBLIC_API_KEY }}
          tmt_plan_regex: "timeout_plan"
          pull_request_status_name: "Timeout test"
          update_pull_request_status: "true"
          timeout: 60
          create_issue_comment: "true"

      - name: Check if testing farm test reached past the timeout limit
        if: ${{ !cancelled() }}
        run: |
          curl ${{ steps.tf_results.outputs.test_log_url }} > results.log
          ret_val=$(grep "This should never happened because of timeout set on GHA" results.log)
          if [[ $ret_val == 0 ]]; then
            echo "Timeout parameter isn't working as expected"
            exit 1
          fi

      - name: Check if timeout was reached
        if: ${{ !cancelled() }}
        run: |
          if [[ ${{ steps.tf_results.conclusion }} == 'success' ]]; then
            echo "Action passed, timeout wasn't reached"
            exit 1
          fi
          echo "Action failed, timeout was reached"
